regression-tests.dnsdist/test_Lua.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+
+import base64
+import time
+import unittest
+from dnsdisttests import DNSDistTest
+
+class TestLuaThread(DNSDistTest):
+    _consoleKey = DNSDistTest.generateConsoleKey()
+    _consoleKeyB64 = base64.b64encode(_consoleKey).decode('ascii')
+
+    _config_params = ['_consoleKeyB64', '_consolePort']
+    _config_template = """
+    setKey("%s")
+    controlSocket("127.0.0.1:%s")
+
+    newThread([==[
+      local socket = require'socket'
+      local i=1
+      while true
+      do
+          socket.sleep(1)
+          submitToMainThread("setCounter", {i=i})
+          i = i + 1
+      end
+    ]==])
+
+    counter = 0
+    function threadmessage(cmd, data)
```

Comment:
Just a nit but perhaps it would be better to define that function before calling `newThread()`? It does not matter here before the background thread will sleep for one second which should be enough for us to return, though.

Reply:
I'll fix that

Reply:
pushed